rskj-core/src/main/java/co/rsk/NodeRunnerImpl.java

```diff
@@ -145,7 +145,11 @@ public synchronized void stop() {
         logger.info("Shutting down RSK node");
 
         for (int i = internalServices.size() - 1; i >= 0; i--) {
-            internalServices.get(i).stop();
+            try {
+                internalServices.get(i).stop();
```

Comment:
If the stop of any of the services fails throwing an exception the execution would stop and the missing services won't stop nicely.

Reply:
ok, I see - so this is not required by the feature itself

I guess it's a tradeoff: non-properly finished service may sometimes result to data inconsistency in next services. We should be very careful with such changes in shutdown process

Reply:
(previous deleted reply) If the stop of any of the services fails throwing an exception the execution would stop and the missing services won't stop nicely.

Reply:
> ok, I see - so this is not required by the feature itself
> 
> I guess it's a tradeoff: non-properly finished service may sometimes result to data inconsistency in next services. We should be very careful with such changes in shutdown process

Huh if it could happen I think it is something we should fix. A wrong/different ending of one service shouldn't affect to the others

Reply:
if that happens then there's a bug in the code which has to be fixed. We don't expect `service.stop()` to throw any checked exception. Only unchecked exceptions can happen here (eg. RuntimeException), and if that happens we should propagate it call stack up. Not sure it's a good idea to catch an exception and continue stopping other services as node is already in inconsistent state - best we can do here is to catch it at upper level and log it. So I'd rather not to add this `try-catch` in here

Reply:
ok got it, I'll remove it. But In my opinion the services should be independent and in case there are dependencies we could create inner groups and close them together. Maybe is something we can try to do in the future. 
As an example, in the case of this service. If the service throws an error due to the file persist or any other thing processing an unexpected input data or so, the others services shouldn't be affected and should be stopped properly.

Reply:
yeah, that was my point. If a service for any reason throws an [unchecked](https://www.baeldung.com/java-checked-unchecked-exceptions#unchecked) exception, which means that the node most probably cannot recover from that state. If the service can recover from an exception that was thrown in the `stop()` method, then it should not be propagated up but rather handled in it.